tests/e2e/specs/in-game/ranked-matches.spec.js

```diff
@@ -0,0 +1,82 @@
+const dayjs = require('dayjs');
+import { setupGameAsP0, setupGameAsP1, Card } from '../../support/helpers';
+import { seasonFixtures } from '../../fixtures/statsFixtures';
+import { playerOne, playerTwo } from '../../fixtures/userFixtures';
+
+describe('Creating And Updating Ranked Matches', () => {
+  beforeEach(function () {
+    cy.wipeDatabase();
+    cy.visit('/');
+    // Sign up to players and store their id's for comparison to match data
+    cy.signupOpponent(playerOne.username, playerOne.password).as('playerOneId');
+    cy.signupOpponent(playerTwo.username, playerTwo.password).as('playerTwoId');
+    // Log in as playerOne
+    cy.loginPlayer(playerOne.username, playerOne.password);
+    setupGameAsP0(true, true);
+    // Set up season
+    const [_clubsSeason, diamondsSeason] = seasonFixtures;
+    diamondsSeason.startTime = dayjs().subtract(2, 'week').subtract(1, 'day').valueOf();
+    diamondsSeason.endTime = dayjs().add(11, 'weeks').valueOf();
+    cy.loadSeasonFixture([diamondsSeason]);
+  });
+  it.only('Creates a match when two players play a ranked game for the first time this week', function () {
```

Comment:
Initially I planned on making many tests for this but now I'm thinking one happy flow test (this) plus a couple edge cases like skipping unranked games or completed matches

Reply:
This is where I landed with a lot of the test cases. Generally one happy path makes the most sense to me, with a few known edge-cases sprinkled in to keep down complexity. Might be worth looking in to this more when we move to Cypress 10.